lib/Parser/Parse.cpp

```diff
@@ -5040,6 +5040,15 @@ ParseNode * Parser::ParseFncDecl(ushort flags, LPCOLESTR pNameHint, const bool n
         }
     }
 
+    if (buildAST && fDeclaration)
+    {
+        Symbol* funcSym = pnodeFnc->GetFuncSymbol();
+        if (funcSym->GetIsFormal())
+        {
+            GetCurrentFunctionNode()->SetHasAnyWriteToFormals(true);
```

Comment:
it is interesting to see not all the places (where we do SetHasAnyWriteToFormals) we check for buildAST (which we should).

Reply:
Until proven that's an issue, not making that change with this change.